src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtHistoryService.java

```diff
@@ -0,0 +1,149 @@
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+
+/**
+ * Service for admin court history.
+ */
+@Service
+@Slf4j
+public class AdminCourtHistoryService {
+
+    private final CourtHistoryRepository courtHistoryRepository;
+
+    private final AdminAuditService adminAuditService;
+
+    @Autowired
+    public AdminCourtHistoryService(CourtHistoryRepository courtHistoryRepository, AdminAuditService adminAuditService) {
+        this.courtHistoryRepository = courtHistoryRepository;
+        this.adminAuditService = adminAuditService;
+    }
+
+    /**
+     * Get all court histories.
+     *
+     * @return {@link List} of {@link CourtHistory} list of all court histories
+     */
+    public List<CourtHistory> getAllCourtHistory() {
+        return courtHistoryRepository.findAll()
+            .stream()
+            .map(CourtHistory::new)
+            .toList();
+    }
+
+    /**
+     * Get a specific court history using ID.
+     *
+     * @param courtHistoryId ID of a court history
+     * @return CourtHistory a specific court history
+     */
+    public CourtHistory getCourtHistoryById(Integer courtHistoryId) {
+        return courtHistoryRepository.findById(courtHistoryId)
+            .map(CourtHistory::new)
+            .orElseThrow(() -> new NotFoundException("Court History with ID: " + courtHistoryId));
+    }
+
+    /**
+     * Get all the court histories of a specific court using a search court ID.
+     *
+     * @param courtId ID of a court
+     * @return {@link List} of {@link CourtHistory} list of the court histories of a specific court
+     */
+    public List<CourtHistory> getCourtHistoryByCourtId(Integer courtId) {
+        return courtHistoryRepository.findAllBySearchCourtId(courtId)
+            .stream()
+            .map(CourtHistory::new)
+            .toList();
+    }
+
+    /**
+     * Get all the court histories matching the court name.
+     *
+     * @param courtName an old court name
+     * @return {@link List} of {@link CourtHistory} list of court histories with matching name
+     */
+    public List<CourtHistory> getCourtHistoryByCourtName(String courtName) {
+        return courtHistoryRepository.findAllByCourtName(courtName)
+            .stream()
+            .map(CourtHistory::new)
+            .toList();
+    }
+
+    /**
+     * Add a court history.
+     * Also saves audit info of the court history creation.
+     * @param courtHistory old court info
+     * @return CourtHistory the new court history that has been added
+     */
+    public CourtHistory addCourtHistory(CourtHistory courtHistory) {
+
+        CourtHistory courtHistoryModel = new CourtHistory(courtHistoryRepository.save(new uk.gov.hmcts.dts.fact.entity.CourtHistory(courtHistory)));
+        adminAuditService.saveAudit("Create court history", courtHistoryModel, null, courtHistoryModel.getCourtName());
+        return courtHistoryModel;
```

Comment:
pedantic one, can do this is one line if you want 

return courtHistoryRepository.save(new uk.gov.hmcts.dts.fact.entity.CourtHistory(courtHistory)));
        adminAuditService.saveAudit("Create court history", courtHistoryModel, null, courtHistoryModel.getCourtName());

Reply:
I could be getting confused but does this not make the saveAudit unreachable?